packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -34,6 +44,13 @@ export interface PipelineProps {
    * You can query the generated Stacks using the {@link Pipeline#crossRegionScaffoldStacks} property.
    */
   crossRegionReplicationBuckets?: { [region: string]: string };
+
+  /**
+   * The list of Stages, in order,
+   * to create this Pipeline with.
+   * You can always add more Stages later by calling {@link Pipeline#addStage}.
+   */
+  stages?: cpapi.IStage[];
```

Comment:
It feels like we can provide a nicer initialization API if users didn't have to explicitly create a `Stage`:

```ts
new Pipeline(this, 'my-pipeline', {
  stages: [
    {
      name: 'Source',
      actions: [ sourceAction ]
    },
    {
      name: 'Build',
      actions: [ buildAction ]
    }
  ]
});
```

When the pipeline is initialized, it can implicitly create the `Stage` objects and add the actions to it.

Reply:
Done.